sympy/polys/matrices/ddm.py

```diff
@@ -958,6 +958,130 @@ def lu(a):
 
         return L, U, swaps
 
+    def _fflu(self):
+        """
+        Private method for Phase 1 of fraction-free LU decomposition.
+        Performs row operations and elimination to compute U and permutation indices.
+
+        Returns:
+            U : Upper triangular matrix after elimination.
+            perm (list): Permutation indices for row swaps.
```

Comment:
I don't think that the matrix returned here is upper triangular. It should probably be called something like LU rather than U and it represents the LU decomposition as a single matrix.